src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/PackagePropertyPage/PackageIconValueProvider.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties.Package
+{
+    [ExportInterceptingPropertyValueProvider(PackageIconPropertyName, ExportInterceptingPropertyValueProviderFile.ProjectFile)]
+    internal sealed class PackageIconValueProvider : InterceptingPropertyValueProviderBase
+    {
+        private const string PackageIconPropertyName = "PackageIcon";
+        private const string PackPropertyName = "Pack";
+        private const string PackagePathPropertyName = "PackagePath";
+        private const string NoneItem = "None";
```

Comment:
When referring to an item's name in code, and when we have a corresponding rule file, you can use the rule to define the constant. For example in this case you'd use `None.SchemaName`.

This can make it easier to find places in code that are using a given item type. Also, while it's not likely we'd remove the `None` item's rule, it will cause compile errors when a rule is removed that would would otherwise need.

Reply:
Holy moly. Took me a minute to understand what you were talking about. I thought you mean the rule file related to the Package property page (`PackagePropertyPage.xaml`). Instead, you meant that MSBuild items have a rule file associated with them. Thus, the `None.xaml` file. And, you can access the name (`<Rule Name="None" />`) of that file via `None.SchemaName`, which is what the rule files generate as part of compilation. Got it. Fixed.

Prior to this comment, I didn't know these rule files per item existed.
![image](https://user-images.githubusercontent.com/17788297/127208254-ca55f9ee-a613-4727-9858-a917556bfdcc.png)